ls1intum/Artemis
# [Bugfix] - Added try-catch to avoid login problems due to invalid Regex

### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, editor, instructor, admin) on the test server https://artemistest2.ase.in.tum.de.

### Motivation and Context


Recently severe login problems occurred due to an invalid regex pattern for retrieving Organizations.

### Description

Added a try-catch block which simply logs a warning about the invalid pattern and does not assign Organizations to the user, instead of throwing the internal server error.

### Steps for Testing

1. Create a new Organization with an invalid email-pattern (e.x. *@invalid) **Note that an error will be thrown here**
2. Create a new User (should be successful)
3. **Remember to remove the invalid Organization and the new User
